ext/standard/var.c

```diff
@@ -1308,8 +1320,30 @@ PHP_FUNCTION(serialize)
 		Z_PARAM_ZVAL(struc)
 	ZEND_PARSE_PARAMETERS_END();
 
+	bool data_is_cv = false;
+	if (Z_TYPE_P(struc) == IS_ARRAY
+	 && !(GC_FLAGS(Z_ARRVAL_P(struc)) & IS_ARRAY_IMMUTABLE)
+	 && EG(current_execute_data)
+	 && EG(current_execute_data)->prev_execute_data) {
+		zend_execute_data *execute_data = EG(current_execute_data)->prev_execute_data;
+		if (execute_data->func && ZEND_USER_CODE(execute_data->func->type)) {
+			zend_op_array *func = &execute_data->func->op_array;
+			const zend_op *opline = execute_data->opline;
+			if (func->opcodes < opline) {
+				const zend_op *prev_opline = opline - 1;
+				if ((prev_opline->opcode == ZEND_SEND_VAR || prev_opline->opcode == ZEND_SEND_VAR_EX) && prev_opline->op1_type == IS_CV) {
+					data_is_cv = true;
```

Comment:
This is a hacky test. It looks right, but it leads to failure on ext/standard/tests/serialize/001.phpt with Optimizer and ASAN. May be the real mistake is in SCCP.

master backtrace
``` 
004+     #0 0x1a7de4a in zif_serialize /home/dmitry/php/php-master/ext/standard/var.c:1322
005+     #1 0x27784b8 in ct_eval_func_call /home/dmitry/php/php-master/Zend/Optimizer/sccp.c:825
006+     #2 0x278a3f0 in sccp_visit_instr /home/dmitry/php/php-master/Zend/Optimizer/sccp.c:1681
007+     #3 0x279bdb7 in scdf_solve /home/dmitry/php/php-master/Zend/Optimizer/scdf.c:166
008+     #4 0x27975f7 in sccp_optimize_op_array /home/dmitry/php/php-master/Zend/Optimizer/sccp.c:2410
009+     #5 0x2661d5a in zend_dfa_optimize_op_array /home/dmitry/php/php-master/Zend/Optimizer/dfa_pass.c:1098
010+     #6 0x25951d8 in zend_optimize_script /home/dmitry/php/php-master/Zend/Optimizer/zend_optimizer.c:1527
011+     #7 0x7fa2f97fdc65 in cache_script_in_shared_memory /home/dmitry/php/php-master/ext/opcache/ZendAccelerator.c:1565
012+     #8 0x7fa2f980670a in persistent_compile_file /home/dmitry/php/php-master/ext/opcache/ZendAccelerator.c:2175
```

Reply:
Good idea.
I'm going to send you a patch today.